### DISCUSSION TITLE
Some improvements in water-air interactions

### DISCUSSION THREAD

**Discussion by MaaiG** [FakeFishGames/Barotrauma#8228] [Ideas and suggestions]
As the depth of the dive increases, the hydrodynamic pressure increases, and hence the rate of water flow into the compartment through the hole etc. Oxygen also does not remain constant!

It’s not working now:
**1. The rate of flooding is always constant.**
  The flooding rate should depend on the depth and pressure!

**2. The flooding of the compartment does not affect the oxygen level in it.**
  The air should be let out!
_Related: https://github.com/Regalis11/Barotrauma/issues/7466 !!!_

**3. We can see the water flowing through the holes, but we can’t see the air coming out through the holes.**
  Should be added air bubbles animation coming out of holes or open hatches etc!

**Comment by mygamingaccount** [comment]
Pumps currently push and pull water from a pocket dimension, evidenced by the fact that they draw power when water enters the ballast.

If they were actual pumps and not gates to the water dimension, they would need to do more work to expel the same water to a greater depth.

Pumps do not do any work when water has to enter the ballast: that's just a vent opening to allow the high pressure water to naturally flow in. Make them not consume power when flooding tanks!

**Reply by MaaiG** [reply]
> Make them not consume power when flooding tanks!

Or to consume a small percentage of energy at a time - as a result of simple opening of the electromechanical valve (without water pumping).
